### PR TITLE
Prakhar 1094

### PR DESCRIPTION
**Pull request by prakharcipl**

### CODE REVIEW

File: src/components/BookCard/BookCard.tsx [hunk 1]
```diff
@@ -1,13 +1,43 @@
...
   We would define the structure of the props in the same file as the component.
   Since, this component has the same props as Type Book, we would assign it to it props.
 */
+
 type BookProps = Book;
 
-export const BookCard = (props: BookProps): JSX.Element => {
+export const BookCard = (book: BookProps): JSX.Element => {
+
+  const dispatch: AppDispatch = useDispatch();
```
Comment (atmeshwar-classic) [inline src/components/BookCard/BookCard.tsx:16]: You can also use AppDispatch as an argument to useDispatch
`const dispatch = useDispatch<AppDispatch>()`

File: src/domain/books/BooksPage.tsx [hunk 2]
```diff
@@ -1,7 +1,36 @@
...
+          <div className="books-page-container">
+            {books.map((book: Book) => (
+              <BookCard 
+                {...book} 
+                key={book.id}
+              />
+            ))}
+          </div>
+        )
+      }
```
Comment (atmeshwar-classic) [inline src/domain/books/BooksPage.tsx:33]: This is OK. But there are more than two conditions here i.e. error. Ideally, write this logic in a fn with guard clauses and call it here.

File: src/domain/cart/CartPage.tsx [hunk 3]
```diff
@@ -1,5 +1,33 @@
...
+            ))}
+          <div className="cart-total">
+            <strong className="cart-total-price">
+              amount → ₹{totalPrice(cart)}
+            </strong>
+          </div>
+        </div>
+      )}
+    </div>
+  );
```
Comment (atmeshwar-classic) [inline src/domain/cart/CartPage.tsx:32]: same as above

File: src/domain/books/BooksPage.tsx (hunk 2 above)
Comment (atmeshwar-classic) [reply src/domain/books/BooksPage.tsx:33]: Your JSX should be as clean as possible

File: src/routes/Router.tsx [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+const Router = () => {
+  return (
+    <BrowserRouter>
+      <Nav />
+      <Routes>
+        <Route path="/" element={<BooksPage />} />
+        <Route path="/cart" element={<CartPage />} />
+      </Routes>
+    </BrowserRouter>
```
Comment (atmeshwar-classic) [inline src/routes/Router.tsx:14]: BrowserRouter and Nav should be inside App component. This file is only for routes

**Review by atmeshwar-classic — CHANGES_REQUESTED** [review]
Hi @prakharcipl Please look at the comments I have made.

**Comment by prakharcipl** [comment]
I'll rectify the changes requested @atmeshwar-classic

**Comment by atmeshwar-classic** [comment]
I am closing this PR as the assessment is complete.
Good job @prakharcipl. Just keep in mind the pointers I have shared with you.

**Pull request closed without merging**